private/buf/bufformat/formatter.go

```diff
@@ -1910,7 +1946,7 @@ func (f *formatter) writeMultilineComments(comments ast.Comments) {
 //	}
 func (f *formatter) writeInlineComments(comments ast.Comments) {
 	for i := 0; i < comments.Len(); i++ {
-		if i > 0 {
+		if i > 0 || comments.Index(i).LeadingWhitespace() != "" || f.lastWritten == ';' || f.lastWritten == '}' {
```

Comment:
It seemed more pleasing to the eye to always separate the comment with a space if it follows the end of a declaration (a semicolon or close brace).